build-check-comments.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# Reliable cross-platform comment checker for go files.
+# Uses docker and stable linux distro for reliable grep regex and
+# to make sure we work on Linux and OS X and oh even Windows! :)
+# Main issue is minor incompatibies between grep on various platforms.
+#
+# Linux: set USE_DOCKER=yes to force run in docker
+# All other platforms: uses docker by default. Set USE_DOCKER=no to override.
+#
+# Checks Go code comments if they have trailing period (excludes protobuffers and vendor files).
+# Comments with more than 3 spaces at beginning are omitted from the check, example: '//    - foo'.
+# This does not include top-level commments for funcs and types.
+#
+# Example:
+#   func main() {
+#	    // comment without period, will trigger check
+#	    //comment without leading space, will trigger check.
+#	    // comment without trailing space, will trigger check.
+#	    // good comment, leading space, ends with period, no trailing space.
+#     //     - more than 3 leading spaces, will pass
+#	    app := kingpin.New(filepath.Base(os.Args[0]), "A block storage based long-term storage for Prometheus")
+#   }
+
+
+# Dir of this script
+declare ROOT_DIR=$(cd $(dirname ${BASH_SOURCE}) && pwd)
+declare THIS_SCRIPT="${ROOT_DIR}/$(basename "${BASH_SOURCE}")"
+
+# Image to use if we do docker-based commands, centos is most stable right?
+declare IMAGE="centos:7"
+
+# User can explicitly ask to run in docker
+declare USE_DOCKER=${USE_DOCKER:=""}
+
+# For OS X, always use Docker as we have nasty
+# compat GNU/BSG issues with grep.
+if test "Linux" != "$(uname || true)"
+then
+  # Allow overriding for non-linux platforms
+  if test "no" != "${USE_DOCKER}"
+  then
+    USE_DOCKER="yes"
+  fi
+fi
+
+if test "yes" == "${USE_DOCKER}"
```

Comment:
Maybe check with `-z "${USE_DOCKER}"` ? Or add also `true` as an option but that's just a nit.

Reply:
Adding `true` is a good idea. I probably will do that. I will have a look at `-z` although it might not work in this scenario. I'm not a big fan of these really, as it adds mental strain on me to remember and understand what those shortcut flags mean.

Reply:
Sorry I off course confused those :) I meant `-n` which should match non-empty string.

Reply:
Well perhaps lets steer clear of these `-z` and `-n` flags? For clarity sake?